src/kakarot/instructions/system_operations.cairo

```diff
@@ -245,6 +255,7 @@ namespace SystemOperations {
             calling_context=ctx,
             return_data_len=call_args.ret_size,
             return_data=call_args.return_data,
+            read_only=ctx.read_only,
```

Comment:
we should check here that

```
with_attr error_message("Kakarot: StateModificationError") {
            assert ctx.read_only * call_args.value = 0;
```
right? (CALL is allowed if `not read_only or value == 0`)

Reply:
Yep, will add the assertion before the subcontext construction 👍